pkg/upload/dsym.go

```diff
@@ -36,82 +38,105 @@ func ProcessDsym(
 ) error {
 
 	var buildSettings *ios.XcodeBuildSettings
-	var dsyms *[]*ios.DwarfInfo
 	var plistData *ios.PlistData
 	var uploadOptions map[string]string
 
-	for _, path := range paths {
-		uploadInfo, err := ios.ProcessPathValues(path, dsymPath, projectRoot)
-		if err != nil {
-			return err
+	var dwarfInfo []*ios.DwarfInfo
+	var tempDirs []string
+
+	// Performs an automatic cleanup of temporary directories at the end
+	defer func() {
+		for _, tempDir := range tempDirs {
+			_ = os.RemoveAll(tempDir)
 		}
+	}()
+
+	for _, path := range paths {
 
-		// If projectRoot is not set explicitly, use uploadInfo to set it (where possible)
-		if projectRoot == "" && uploadInfo.ProjectRoot != "" {
-			projectRoot = uploadInfo.ProjectRoot
+		if path == "" {
+			// set path to current directory if not set
+			path, _ = os.Getwd()
 		}
 
-		// If scheme is set explicitly, check if it exists
-		if scheme != "" && len(uploadInfo.DsymPaths) == 0 {
-			_, err = ios.IsSchemeInPath(path, scheme)
-			if err != nil {
-				return err
-			}
-			log.Info("Using scheme: " + scheme)
-		} else {
+		if ios.IsPathAnXcodeProjectOrWorkspace(path) {
+			projectRoot = ios.GetDefaultProjectRoot(path, projectRoot)
+			log.Info("Using '" + projectRoot + "' as project root, this can be changed at any time with a --project-root flag")
```

Comment:
Suggest toning it down a bit to avoid causing alarm, as this is perfectly valid:

```suggestion
			log.Info("Defaulting to '" + projectRoot + "' as the project root")
```